AMM/contracts/AMM/src/events.sw

```diff
@@ -0,0 +1,15 @@
+library events;
+
+use std::contract_id::ContractId;
+
+pub struct DefineValidExchangeEvent {
+    /// The bytecode root of the valid exchange contract implementation
+    root: b256,
+}
+
+pub struct RegisterPoolEvent {
+    /// The pair of asset identifiers that make up the pool
+    pair: (ContractId, ContractId),
+    /// The exchange contract identifier that manages the pool which also identifies the pool asset
```

Comment:
Does `pool asset` here refer to the lp tokens the contract mints?

Reply:
Yes, SwaySwap used the term `token` but it is replaced with `asset` in this application due to the UTXO-based model.